NuProj fails on MSBuild 14.0 and above

On a machine with only MSBuild 14.0, I get this error:

```
error MSB4062: The "Microsoft.Build.Tasks.AssignProjectConfiguration" task could not be
loaded from the assembly Microsoft.Build.Tasks.v12.0, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a. Could not load file or assembly 'Micr
osoft.Build.Tasks.v12.0, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specif
ied. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class t
hat implements Microsoft.Build.Framework.ITask.
```

It looks like [these lines](https://github.com/nuproj/nuproj/blob/master/src/NuProj.Targets/NuProj.targets#L176-L177) mean NuProj will only work if MSBuild 12.0 is installed?

In MSBuild 14.0 they've renamed `Microsoft.Build.Tasks.v12.0.dll` to `Microsoft.Build.Tasks.Core.dll`.  So I'd like to update this .targets file to have logic similar to [this](https://github.com/MobileEssentials/MSBuilder/blob/master/src/CodeTaskAssembly/build/MSBuilder.CodeTaskAssembly.props).  This will also ensure it works with all versions going forward.

See [my fork](https://github.com/nuproj/nuproj/compare/master...jeffkl:dev/jeffkl/msbuild14?expand=1) for the change I'm proposing.  It worked on my machine but I was also wondering if the unit tests would prove it doesn't break anything?  Glancing at them it looks like its covering this kind of change to the .targets.

Thanks

--- a/src/NuProj.Targets/NuProj.targets
+++ b/src/NuProj.Targets/NuProj.targets
@@ -174,7 +174,8 @@
   -->
   
   <UsingTask TaskName="Microsoft.Build.Tasks.AssignProjectConfiguration"            AssemblyName="Microsoft.Build.Tasks.v4.0, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="'$(MSBuildAssemblyVersion)' == ''" />
-  <UsingTask TaskName="Microsoft.Build.Tasks.AssignProjectConfiguration"            AssemblyName="Microsoft.Build.Tasks.v12.0, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="'$(MSBuildAssemblyVersion)' != ''" />
+  <UsingTask TaskName="Microsoft.Build.Tasks.AssignProjectConfiguration"            AssemblyName="Microsoft.Build.Tasks.v$(MSBuildAssemblyVersion), Version=$(MSBuildAssemblyVersion).0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="'$(MSBuildAssemblyVersion)' &lt;= '12.0'" />
+  <UsingTask TaskName="Microsoft.Build.Tasks.AssignProjectConfiguration"            AssemblyName="Microsoft.Build.Tasks.Core, Version=$(MSBuildAssemblyVersion).0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" Condition="'$(MSBuildAssemblyVersion)' &gt;= '14.0'" />
 
   <!--
       CUSTOM TASKS

